Formula/elan-init.rb

```diff
@@ -0,0 +1,49 @@
+class ElanInit < Formula
+  desc "Lean Theorem Prover installer and version manager"
+  homepage "https://leanprover-community.github.io/"
+  url "https://github.com/leanprover/elan/archive/v1.0.3.tar.gz"
+  sha256 "fb3ddd8915a0694ead0f3a51fdf8b0a5540f983f44eee0e757339244c522b8ee"
+  license "Apache-2.0"
+  head "https://github.com/leanprover/elan.git"
+
+  depends_on "rust" => :build
+
+  # elan-init will run on arm64 Macs, but will fetch Leans that are x86_64.
+  depends_on arch: :x86_64
```

Comment:
I'm unsure about including this. Presumably, it fetches an x86_64 Lean because it fetches a pre-built one. Where do those come from?

It seems Lean should build fine on ARM, though, as the `lean` formula has an ARM bottle.

Reply:
You're exactly correct -- it's because it fetches pre-built ones. Upstream there's no pre-built Lean for ARM binaries for macOS so it always fetches x86_64 ones and then breaks (breaks expecting x86_64 libgmp to exist, so not rosetta related). The releases it pulls from are e.g. https://github.com/leanprover-community/lean/releases/tag/v3.30.0 for "stable" and https://github.com/leanprover/lean4/releases/tag/v4.0.0-m2 for lean4 -- and since that's built on GH Actions, there's no ARM build for macOS there (yet).

Lean indeed builds fine on ARM, I do that myself locally too for my own Lean binary -- I just put this in thinking it'd confuse ARM users if they installed this and got non-working Leans.

But let me know what your advice is of course!

Reply:
Ok, if the fetched x86_64 Lean doesn't work, then this is fine.